hls-plugin-api/src/Ide/Plugin/Resolve.hs

```diff
@@ -29,44 +31,61 @@ import           Language.LSP.Server           (LspM, LspT,
+        LogInvalidParamsError re -> pretty $ re ^. L.message
+        LogInternalError re -> pretty $ re ^. L.message
+
+logAndThrow :: Recorder (WithPriority Log) -> (ResponseError -> Log) -> ResponseError -> ExceptT ResponseError (LspT Config IO) a
+logAndThrow recorder logType err = do
+  logWith recorder Error (logType err)
+  throwE err
+
 -- |When provided with both a codeAction provider and an affiliated codeAction
 -- resolve provider, this function creates a handler that automatically uses
 -- your resolve provider to fill out you original codeAction if the client doesn't
 -- have codeAction resolve support. This means you don't have to check whether
 -- the client supports resolve and act accordingly in your own providers.
 mkCodeActionHandlerWithResolve
   :: forall ideState a. (A.FromJSON a) =>
-   (ideState -> PluginId -> CodeActionParams -> LspM Config (Either ResponseError ([Command |? CodeAction] |? Null)))
+  Recorder (WithPriority Log)
+  -> (ideState -> PluginId -> CodeActionParams -> LspM Config (Either ResponseError ([Command |? CodeAction] |? Null)))
   -> (ideState -> PluginId -> CodeAction -> Uri -> a -> LspM Config (Either ResponseError CodeAction))
   -> PluginHandlers ideState
-mkCodeActionHandlerWithResolve codeActionMethod codeResolveMethod =
-    let newCodeActionMethod ideState pid params = runExceptT $
-            do codeActionReturn <- ExceptT $ codeActionMethod ideState pid params
-               caps <- lift getClientCapabilities
-               case codeActionReturn of
-                r@(InR Null) -> pure r
-                (InL ls) | -- If the client supports resolve, we will wrap the resolve data in a owned
-                           -- resolve data type to allow the server to know who to send the resolve request to
-                           supportsCodeActionResolve caps -> pure $ InL ls
-                           --This is the actual part where we call resolveCodeAction which fills in the edit data for the client
-                         | otherwise -> InL <$> traverse (resolveCodeAction (params ^. L.textDocument . L.uri) ideState pid) ls
-    in (mkPluginHandler SMethod_TextDocumentCodeAction newCodeActionMethod
-    <> mkResolveHandler SMethod_CodeActionResolve codeResolveMethod)
-    where
-        dropData :: CodeAction -> CodeAction
+mkCodeActionHandlerWithResolve recorder codeActionMethod codeResolveMethod =
+  let newCodeActionMethod ideState pid params = runExceptT $
+        do codeActionReturn <- ExceptT $ codeActionMethod ideState pid params
+           caps <- lift getClientCapabilities
+           case codeActionReturn of
+             r@(InR Null) -> pure r
+             (InL ls) | -- If the client supports resolve, we will wrap the resolve data in a owned
+                        -- resolve data type to allow the server to know who to send the resolve request to
+                        supportsCodeActionResolve caps -> pure $ InL ls
```

Comment:
Somewhere around here we might want to log at the lowest level to record what decision we're making about whether or not to try and use resolve?